### ISSUE TITLE
Unable to Get How its Mesh Invariant??

### ISSUE THREAD

**Issue by Karn3003** [bogdanraonic3/ConvolutionalNeuralOperator#1]
Hi,
I went through the paper and codebase. I have doubts about how it is invariant to mesh size if we train on 48x48 then how to transfer the solution to 85x85 like superresolution in FNO??

**Comment by bogdanraonic3** [comment]
Hi Karn, 

In the paper SM C.4, we metnioned that

_It is important to note that the implemented CNO models are specified on a predefined computational grid with a sampling rate of s ≥ 2w. The input functions must be compatible with this grid. If the input function is not compatible with the computational grid, one needs transform it to an appropriate representation (downsampling/usampling). Once the model is applied, the output is transformed back to the original representation._

Please refer to [https://arxiv.org/pdf/2302.01178.pdf](url) for more details.

**Comment by Karn3003** [comment]
Yeah, that helped. So If I have enough sampling rate (s ≥ 2w) and I have trained the model on a certain mesh size for that sampling rate so to transfer the solution I need to do a transformation such that it should be of the same size and later transform it back to input shape. Am I right??

**Comment by bogdanraonic3** [comment]
The computational grid is the grid that you perform all the operations on (i.e. training grid). All the training functions are sampled on that grid. If you want to evaluate model on different grid, first you have resample it (or transfer it) to the computational grid, apply the model, and then sample/transfer it back to the desired grid. Resampling can be done by appling up/downsampling directly in the Fourier space or using windowed-sinc filters.

This mechanism works both for higher and lower resolutions.

**Comment by Karn3003** [comment]
Alright Thanks.

**Comment by Karn3003** [comment]
Hi, 
One more question I have is whether there are any resources where I can refer to the following arguments in the synthesis Layer. 

is_critically_sampled,          # Does this layer use critical sampling?  #NOT IMPORTANT FOR CNO.

        in_channels,                    # Number of input channels.
        out_channels,                   # Number of output channels.
        in_size,                        # Input spatial size: int or [width, height].
        out_size,                       # Output spatial size: int or [width, height].
        in_sampling_rate,               # Input sampling rate (s).
        out_sampling_rate,              # Output sampling rate (s).
        in_cutoff,                      # Input cutoff frequency (f_c).
        out_cutoff,                     # Output cutoff frequency (f_c).
        in_half_width,                  # Input transition band half-width (f_h).
        out_half_width,                 # Output Transition band half-width (f_h).

        # Hyperparameters.
        filter_size         = 6,        # Low-pass filter size relative to the lower resolution when up/downsampling.
        lrelu_upsampling    = 2,        # Relative sampling rate for leaky ReLU. Ignored for final the ToRGB layer.
        use_radial_filters  = False,     # Use radially symmetric downsampling filter? Ignored for critically sampled layers.

**Comment by Karn3003** [comment]
Hi @bogdanraonic3 ,
I saw the plot where you have provided the error across different resolutions between FNO and CNO. But I don't think it's a fair comparison between both methods as for CNO across different dimensions you are resampling the signal in the Fourier domain but for FNO you are directly passing through the FNO as it is invariant to mesh size. But you should also resample the signal in the Fourier domain for FNO and plot test error which will be a fair comparison.
